api/odk/helpers/aggregate-submissions.js

```diff
@@ -36,6 +62,13 @@ module.exports = function (formName, errorCallback, aggregateCallback) {
             });
             return;
         }
+
+        // if offset, we do pagination
+        if (typeof offset !== 'undefined' && offset !== null) {
+            submissionDirs = submissionDirs.slice(offset, offset + limit);
+        }
+
+        // no results
         if (submissionDirs.length === 0) {
             aggregateCallback([]);
```

Comment:
No need to short-circuit, since `submissionDirs == []` at this point, and `eachLimit` will skip straight to the callback.